### PR TITLE
Add Ruby in run-code

### PR DESCRIPTION
**Pull request by WangGithub0**
image

### CODE REVIEW

**Comment by tarasglek** [comment]
lgtm, love how easy these are to add

File: src/lib/run-code.ts [hunk 1]
```diff
@@ -148,6 +158,39 @@ async function runPython(code: string) {
   return executionResult;
 }
 
+async function runRuby(code: string) {
```
Comment (humphd) [inline src/lib/run-code.ts:161]: This is nearly identical to the python code, can we refactor to share common bits?

File: src/lib/run-code.ts [hunk 2]
```diff
@@ -148,6 +158,39 @@ async function runPython(code: string) {
...
+        stdout: (out) => {
+          console.log(out);
+        },
+        stderr: (err) => {
+          console.error(err);
+        },
+      })
+        .then((result) => {
+          if (result.exitCode === 0) {
+            resolve(); // Resolves the promise with no value
```
Comment (humphd) [inline src/lib/run-code.ts:179]: This comment is not needed.

File: src/lib/run-code.ts [hunk 3]
```diff
@@ -148,6 +158,39 @@ async function runPython(code: string) {
...
+          } else {
+            reject(new Error("Script execution failed")); // Rejects the promise if execution failed
```
Comment (humphd) [inline src/lib/run-code.ts:181]: Same here.

File: src/lib/run-code.ts [hunk 4]
```diff
@@ -148,6 +158,39 @@ async function runPython(code: string) {
...
+          }
+        })
+        .catch((error) => {
+          reject(error); // Reject on error
```
Comment (humphd) [inline src/lib/run-code.ts:185]: And here.  Let the code speak for itself if you have nothing more to add.

**Review by humphd — CHANGES_REQUESTED** [review]
This is really clean.  A few things to figure out.

File: src/lib/run-code.ts [hunk 5]
```diff
@@ -115,15 +125,25 @@ async function runJavaScript(code: string) {
   }
 }
 
-async function runPython(code: string) {
+async function runPythonOrRuby(code: string, language: string) {
```
Comment (humphd) [inline src/lib/run-code.ts:128]: `runInWasi()` would be a better name.

**Review by humphd — CHANGES_REQUESTED** [review]
Looks good, one thing to fix.

**Review by kliu57 — APPROVED** [review]
Tested it in browser, it works as expected!

**Pull request merged**